### PR TITLE
Add paper background support to side navigation

### PR DESCRIPTION
**Pull request by bartaz**
## Done

As part one building block of new documentation layout this PR updates side navigation component by adding new paper background colour support to the light theme.

## QA

- Open [demo](https://vanilla-framework-4881.demos.haus/docs/examples/patterns/side-navigation/paper)
- Make sure side navigation looks fine on paper background (check mobile view as well).
  - example contains 3 variants (standard, with icons and sticky)

### Check if PR is ready for release

If this PR contains Vanilla SCSS code changes, it should contain the following changes to make sure it's ready for the release:

- [x] PR should have one of the following labels to automatically categorise it in release notes:
  - `Feature 🎁`, `Breaking Change 💣`, `Bug 🐛`, `Documentation 📝`, `Maintenance 🔨`.
- [x] Vanilla version in `package.json` should be updated relative to the [most recent release](https://github.com/canonical/vanilla-framework/releases/latest), following semver convention:
  - if CSS class names are not changed it can be bugfix relesase (x.x.**X**)
  - if CSS class names are changed/added/removed it should be minor version (x.**X**.0)
  - see the [wiki for more details](https://github.com/canonical/vanilla-framework/wiki/Release-process#pre-release-tasks)
- [x] Any changes to component class names (new patterns, variants, removed or added features) should be listed on the [what's new page](https://github.com/canonical/vanilla-framework/blob/main/releases.yml).

### CODE REVIEW

File: scss/_patterns_side-navigation.scss [hunk 1]
```diff
@@ -539,7 +539,7 @@
     .p-side-navigation--raw-html {
       @include vf-side-navigation-raw-html-theme-light;
 
-      &.is-light {
+      &.is-dark {
```
Comment (bartaz) [inline scss/_patterns_side-navigation.scss:542]: Drive-by: fixed a broken class name when switching from light theme to dark theme in raw HTML variant.

File: tests/parker.js [hunk 2]
```diff
@@ -12,7 +12,7 @@ function generateMetrics(file, metricsArray) {
     {
       name: 'Stylesheet size',
       benchmark: 150000,
-      threshold: 380000,
+      threshold: 430000,
```
Comment (bartaz) [inline tests/parker.js:15]: Adding more themes to side navigation increases the size of CSS noticeably. 
It's OK, because in the end this additional CSS is quite repetitive and will GZIP nicely, so doesn't affect performance too much.

But we need to increase our allowed thresholds so that the tests pass.

**Comment by andesol** [comment]
@bartaz Looks like the second one (with icons) has a transparent background on mobile 

image

**Comment by bartaz** [comment]
> @bartaz Looks like the second one (with icons) has a transparent background on mobile

Thanks @andesol, good catch. I messed up some overrides in example HTML. Should be good now.

**Review by andesol — APPROVED** [review]
LGTM :+1:
